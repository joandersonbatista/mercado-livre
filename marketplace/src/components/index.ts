export { Header } from './header/header';
export { IconButton } from './icon-button/icon-button';
export { Button } from './button/button';
export { Breadcrumbs } from './breadcrumbs/breadcrumbs';
export { BreadcrumbsSkeleton } from './breadcrumbs/breadcrumbs-skeleton';
export { Skeleton, type SkeletonProps } from './skeleton/skeleton';
export { LanguageSwitcher } from './language-switcher/language-switcher';
export { ServerError } from './server-error/server-error';
