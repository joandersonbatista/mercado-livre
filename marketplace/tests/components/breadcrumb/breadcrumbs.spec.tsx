import { Breadcrumbs } from '@/components';
import { render, screen } from '@testing-library/react';

describe('Breadcrumbs Component', () => {
  it('should render breadcrumbs items correctly', () => {
    const items = [{ name: 'Home' }, { name: 'Category' }, { name: 'Product' }];

    render(<Breadcrumbs items={items} />);

    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Category')).toBeInTheDocument();
    expect(screen.getByText('Product')).toBeInTheDocument();
  });

  it('should display chevron icon between breadcrumbs items', () => {
    const items = [{ name: 'Home' }, { name: 'Category' }, { name: 'Product' }];

    render(<Breadcrumbs items={items} />);

    const chevrons = screen.getAllByAltText('icone chevron');
    expect(chevrons.length).toBe(2);
  });

  it('should not display chevron after the last breadcrumbs item', () => {
    const items = [{ name: 'Home' }, { name: 'Product' }];

    render(<Breadcrumbs items={items} />);

    const chevrons = screen.queryAllByAltText('icone chevron');
    expect(chevrons.length).toBe(1);
  });

  it('should render correct link for each breadcrumbs item', () => {
    const items = [{ name: 'Home' }, { name: 'Category' }, { name: 'Product' }];

    render(<Breadcrumbs items={items} />);

    const links = screen.getAllByRole('link');
    expect(links[0]).toHaveTextContent('Home');
    expect(links[1]).toHaveTextContent('Category');
    expect(links[2]).toHaveTextContent('Product');
  });

  it('should match the snapshot', () => {
    const items = [{ name: 'Home' }, { name: 'Category' }, { name: 'Product' }];

    const { container } = render(<Breadcrumbs items={items} />);
    expect(container).toMatchSnapshot();
  });
});
