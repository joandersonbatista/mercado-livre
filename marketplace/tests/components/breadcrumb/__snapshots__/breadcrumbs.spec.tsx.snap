// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs Component should match the snapshot 1`] = `
<div>
  <nav
    class="nav-breadcrumbs"
  >
    <ul>
      <li>
        <a
          href=""
          role="link"
        >
          Home
        </a>
        <img
          alt="icone chevron"
          data-nimg="1"
          decoding="async"
          height="8"
          loading="lazy"
          src="/icons/chevron.svg"
          style="color: transparent;"
          width="10"
        />
      </li>
      <li>
        <a
          href=""
          role="link"
        >
          Category
        </a>
        <img
          alt="icone chevron"
          data-nimg="1"
          decoding="async"
          height="8"
          loading="lazy"
          src="/icons/chevron.svg"
          style="color: transparent;"
          width="10"
        />
      </li>
      <li>
        <a
          href=""
          role="link"
        >
          Product
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
